presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -46,15 +53,31 @@
         private final PlanNodeId planNodeId;
         private final Step step;
         private final List<AccumulatorFactory> accumulatorFactories;
+        private final List<Type> sourceTypes;
```

Comment:
Do we need to amend `AggregationOperator` at all? I think the best would be to use some kind of `SortOperator` as input for regular `AggregationOperator`. IMO composition is more preferable.